File: Leetcode/1636.cpp
/**
 * 1636. 按照频率将数组升序排序
 * link：https://leetcode-cn.com/problems/sort-array-by-increasing-frequency/
 */

class Solution
{
public:
    vector<int> frequencySort(vector<int> &nums)
    {
        int count[201] = {0};
        for (const int &num : nums)
            count[num + 100]++;
        for (int i = 0; i < nums.size(); i++)
            nums[i] = 1000 * count[nums[i] + 100] + 100 - nums[i];
        sort(nums.begin(), nums.end());
        for (int &num : nums)
            num = 100 - (num % 1000);
        return nums;
    }
};

File: Leetcode/1636.js
/**
 * 1636. 按照频率将数组升序排序
 * link：https://leetcode.cn/problems/sort-array-by-increasing-frequency/
 */

/**
 * @param {number[]} nums
 * @return {number[]}
 */
var frequencySort = function (nums) {
    const count = new Map();
    for (const num of nums) {
        const freq = count.get(num) ?? 0;
        count.set(num, freq + 1);
    }
    return nums.sort((a, b) => {
        const freqA = count.get(a);
        const freqB = count.get(b);
        if (freqA === freqB) {
            return b - a;
        }
        return freqA - freqB;
    });
};